test/cpp/end2end/generic_end2end_test.cc

```diff
@@ -330,6 +381,32 @@ TEST_F(GenericEnd2endTest, Deadline) {
                        gpr_time_from_seconds(10, GPR_TIMESPAN)));
 }
 
+TEST_F(GenericEnd2endTest, ShortDeadline) {
+  ResetStub();
+
+  ClientContext cli_ctx;
+  EchoRequest request;
+  EchoResponse response;
+
+  {
+    std::lock_guard<std::mutex> lock(shutting_down_mu_);
```

Comment:
You shouldn't need to lock at this point since there's no concurrent thread yet.

Reply:
Done